File: frontend/classes/cordinate.php
<?php
/**
 *		Degree:
 *		-------------
 *		Longitude:Y
 *		E [ 000-179 ] +
 *		W [ 001-180 ] -
 *		
 *		Latitude:X
 *		N [00 - 89] + 
 *		S [01 - 90] -
 *		
 *		Minute, Second:
 *		------------
 *		[00-59]
 *
 */

class Cordinate {
		
	public $deg;
	public $min;
	public $sec;
		
	public $rad;
	public $dec;
		
	function __construct($deg, $min, $sec) {
		$this->deg = $deg;
		$this->min = $min;
		$this->sec = $sec;			
	}
		
	/**
	*	Converts DMS ( Degrees / minutes / seconds ) 
	*	to decimal format longitude / latitude
	**/
	public function DMStoDEC($deg, $min, $sec) {
		return $deg + ((($min*60)+($sec))/3600);
	}

	/**
	* Converts decimal longitude / latitude to DMS
	* ( Degrees / minutes / seconds ) 
	*
	**/
	public function DECtoDMS($dec) {
		// Converting float to integer
		$deg = (int) $dec;
		$rem = $dec-$deg;
		$rem *= 3600;
		$min = floor($rem / 60);
		$sec = $rem - ($min*60);
		return array($deg, round($min), round($sec));
	}				
}
?>

File: radio/src/radio/Cordinate.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package radio;
import static java.lang.Math.*;
/**
 *
 * @author egidijus
 */
public class Cordinate {		
	public int deg;
	public int min;
	public int sec;
		
	public double rad;
	public double dec;
		
	public void Cordinate(int deg, int min, int sec) {
		this.deg = deg;
		this.min = min;
		this.sec = sec;			
	}
		
	/**
	* Converts DMS ( Degree / minute / second ) to decimal format longitude / latitude
        * @param deg degree
        * @param min minute
        * @param sec second
        * @return longitude / latitude in decimal format
	**/
	static double DMStoDEC(int deg, int min, int sec) {
           return deg + ((double)min*60+sec)/3600;
	}
        
        static short[] DECtoDMS(double dec) {
          
            short deg = (short) dec;
            double rem = dec-deg;
            rem *= 3600;
            short min = (short) (rem / 60);
            short sec = (short) (rem - (min*60));
            return new short[]{deg, min, sec};
	}	

}
